Assignment2/Q1/src/BinaryTree.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Created by Alice Colt.
+ *
+ * Implementaiton of a Binary Tree with functions for insert, serach and takeAncestors
+ *
+ */
+public class BinaryTree {
+    private NodeBinaryTree root;
+
+    //Constructor
+    public BinaryTree(NodeBinaryTree new_root){
+        root = new_root;
+    }
+
+
+    public void insert(NodeBinaryTree new_node) {
```

Comment:
It would also be nice if this code coped with inserting a node into an empty tree. Requiring a tree start with a single element and adding the rest isn't quite as neat. However, since the tree-construction part isn't the main point of this exercise, I'm not too worried.